src/plone/api/addon.py

```diff
@@ -0,0 +1,283 @@
+            }
+            install_profile = installer.get_install_profile(product_id)
+            if install_profile is not None:
+                product["title"] = install_profile["title"]
+                product["description"] = install_profile["description"]
+                product["install_profile"] = install_profile
+                product["profile_type"] = "default"
+            uninstall_profile = installer.get_uninstall_profile(product_id)
+            if uninstall_profile is not None:
+                product["uninstall_profile"] = uninstall_profile
+                # Do not override profile_type.
+                if not product["profile_type"]:
+                    product["profile_type"] = "uninstall"
+        if "version" in profile:
+            product["upgrade_profiles"][profile["version"]] = profile
+        else:
+            product["other_profiles"].append(profile)
+        addons[product_id] = AddonInformation(**product)
+    return tuple(addons.items())
+
+
+def _update_addon_info(
+    installer: InstallerView, addon: AddonInformation
+) -> AddonInformation:
+    """Update information about an addon."""
+    addon_id = addon.id
+    if addon.valid:
+        flags = []
+        # Update only what could be changed
+        is_installed = installer.is_product_installed(addon_id)
+        if is_installed:
+            addon.upgrade_info = installer.upgrade_info(addon_id) or {}
+            if addon.upgrade_info.get("available"):
+                flags.append("upgradable")
+            else:
+                flags.append("installed")
+        else:
+            flags.append("available")
+        addon.flags = flags
+    return addon
+
+
+def _get_addons() -> List[AddonInformation]:
+    """Return an updated list of addon information.
+
+    :returns: List of AddonInformation.
+    """
+    installer = _get_installer()
+    addons = dict(_cached_addons())
+    result = []
+    for addon in addons.values():
+        result.append(_update_addon_info(installer, addon))
+    return result
+
+
+def get_addons(limit: str = "") -> List[AddonInformation]:
+    """List addons in this Plone site.
+
+    :param limit: Limit list of addons.
+        'installed': only products that are installed and not hidden
+        'upgradable': only products with upgrades
+        'available': products that are not installed but could be
+        'non_installable': Non installable products
+        'broken': uninstallable products with broken dependencies
+    :returns: List of AddonInformation.
+    """
+    addons = _get_addons()
+    if limit in ("non_installable", "broken"):
+        return [addon for addon in addons if limit in addon.flags]
+
+    addons = [addon for addon in addons if addon.valid]
+    if limit in ("installed", "upgradable", "available"):
+        addons = [addon for addon in addons if limit in addon.flags]
+    elif limit != "":
+        raise InvalidParameterError(f"Value {limit} for parameter mode is not valid.")
```

Comment:
```suggestion
        raise InvalidParameterError(f"Parameter {limit=} is not valid.")
```

(will print e.g. "Parameter limit=bogus is not valid.")